apis/core/bootstrap/index.ts

```diff
@@ -0,0 +1,25 @@
+import $rdf from 'rdf-ext'
+import Parser from '@rdfjs/parser-n3'
+import toStream from 'string-to-stream'
+import StreamClient from 'sparql-http-client/StreamClient'
+import { cubes } from './cubes'
+import { log } from '../lib/log'
+import { entrypoint } from './entrypoint'
+
+const parser = new Parser()
+const initialResources = [cubes, entrypoint]
+
+export async function bootstrap(storeUrl: string, base: string) {
+  const client = new StreamClient({
+    storeUrl,
+  })
+
+  const dataset = await initialResources.reduce((previous, turtle) => {
```

Comment:
I don't understand this part. `cubes` and `entrypoint` are a dataset right? You are transforming the `dataset` into a `string`, then a `stream`, and then parsing it again into a `stream of quads`, and finally importing it into another `dataset`?

Reply:
`cubes` and `entrypoint` are just strings. They are all reduced-parsed into the `dataset` and then pushed to the store

Reply:
Do I understand correctly: every time we boot up the app, it will actually insert data in the triplestore?

How does it handle upgrades? What if the data shape changes? Does it replace the data every time?

Reply:
Yes, simply replace those resources every time.

Reply:
Ok thanks!